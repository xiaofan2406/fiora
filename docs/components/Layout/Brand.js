/* @flow */
import React from 'react';
import { css } from 'react-emotion';
import { fontSizes, spacing } from 'styles';
import { APP_TITLE } from 'utils/constants';
import logo from 'assets/logo.svg';

const cssBrand = css`
  display: flex;
  align-items: center;
  padding: ${spacing.internal}px;
  & > .logo {
    animation: spin infinite 10s linear;
    height: 36px;
  }
  & > .title {
    animation: fadeIn 2s ease;
    font-size: ${fontSizes.large}px;
  }
`;

class Brand extends React.Component<{}> {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <div className={cssBrand}>
        <img src={logo} alt="logo" className="logo" />
        <span className="title">{APP_TITLE}</span>
      </div>
    );
  }
}

export { Brand as Component };

export default Brand;
