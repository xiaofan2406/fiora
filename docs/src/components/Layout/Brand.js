import React from 'react';
import { css } from 'emotion';
import { variables, fontSizes, theme } from 'styles';
import { APP_TITLE } from 'configs';
import logo from 'assets/logo.svg';

const brand = css`
  display: flex;
  align-items: center;
  & img {
    animation: spin infinite 10s linear;
    height: ${variables.Brand.logoHeight}px;
  }
  & span {
    animation: fadeIn 2s ease;
    font-size: ${fontSizes.large}px;
    color: ${theme.inverseColor};
  }
`;

class Brand extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <div className={brand}>
        <img src={logo} alt="logo" />
        <span>{APP_TITLE}</span>
      </div>
    );
  }
}

export { Brand as Component };

export default Brand;
