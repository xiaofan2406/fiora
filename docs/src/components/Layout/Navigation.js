import React from 'react';
import { css } from 'emotion';
import { NavLink } from 'react-router-dom';
import { ROUTES } from 'configs';
import { theme, variables } from 'styles';

const navLink = css`
  color: ${theme.inverseColor};
  text-decoration: none;
  padding: 0 0.5em;
  display: inline-block;
  line-height: ${variables.Layout.headerHeight}px;
  height: ${variables.Layout.headerHeight}px;
  &:hover {
    background-color: ${theme.bgAccentColor};
  }
`;

const navLinkActive = css`
  border-bottom: 2px solid ${theme.primaryColor};
`;

function Navigation() {
  return (
    <div>
      {Object.values(ROUTES).map(route => (
        <NavLink
          className={navLink}
          activeClassName={navLinkActive}
          key={route.path}
          exact={route.exact}
          to={route.path}
        >
          {route.name}
        </NavLink>
      ))}
    </div>
  );
}

export { Navigation as Component };

export default Navigation;
