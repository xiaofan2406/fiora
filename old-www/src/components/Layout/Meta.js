import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

function Meta({ metadata }) {
  return (
    <Helmet
      title={metadata.title}
      meta={[
        { name: 'description', content: metadata.description },
        {
          name: 'keywords',
          content: metadata.keywords.join(', '),
        },
      ]}
    >
      <html lang="en" />
    </Helmet>
  );
}

Meta.propTypes = {
  metadata: PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    keywords: PropTypes.array.isRequired,
  }).isRequired,
};

export default Meta;
