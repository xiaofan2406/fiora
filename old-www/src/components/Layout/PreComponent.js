import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'react-emotion';
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live';
import createFiora from '../../../../src';

const cssLive = css`
  display: flex;
  justify-content: space-between;
  margin-bottom: 24px;
`;

const cssEditor = css`
  width: 480px;
  padding: 24px;
  position: relative;
`;

const cssPreview = css`
  padding: 24px;
  width: 396px;
  background-color: #fafafa;
`;

class PreComponent extends React.Component {
  instance = createFiora();

  render() {
    const { children } = this.props;
    const showInteractive =
      children.props.props &&
      !children.props.props.className.endsWith('-static');

    return (
      <LiveProvider
        code={children.props.children}
        scope={this.instance}
        mountStylesheet={false}
        className={showInteractive ? cssLive : ''}
      >
        <LiveEditor
          contentEditable={showInteractive}
          className={showInteractive ? cssEditor : ''}
        />
        {showInteractive ? <LiveError /> : null}
        {showInteractive ? <LivePreview className={cssPreview} /> : null}
      </LiveProvider>
    );
  }
}

PreComponent.propTypes = {
  children: PropTypes.node.isRequired,
};

export default PreComponent;
