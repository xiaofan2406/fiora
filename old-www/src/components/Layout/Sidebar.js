import React from 'react';
import { Link } from 'gatsby';
import PropTypes from 'prop-types';
import { css } from 'react-emotion';

const cssSidebar = css`
  position: fixed;
  left: 0;
  width: 240px;
  height: 100vh;
  background: #f04242;
  display: flex;
  flex-direction: column;
`;

const cssNav = css`
  flex: 1;
`;

const cssTitle = css`
  margin-top: 12px;
  font-size: 54px;
  color: #fff;
  text-align: center;
  cursor: default;
`;

const cssLink = css`
  color: #fff;
  display: block;
  font-size: 18px;
  font-weight: 200;
  padding: 4px 24px;
  text-decoration: none;
  &:hover {
    background-color: rgba(255, 255, 255, 0.24);
  }
`;

const activeLinkStyle = {
  backgroundColor: ' rgba(255, 255, 255, 0.24)',
};

function Sidebar({ siteTitle }) {
  return (
    <div className={cssSidebar}>
      <div className={cssNav}>
        <h1 className={cssTitle}>{siteTitle}</h1>
        <Link to="/" className={cssLink} activeStyle={activeLinkStyle}>
          Get Started
        </Link>
        <Link to="/tutorial" className={cssLink} activeStyle={activeLinkStyle}>
          Tutorial
        </Link>
        <Link to="/api" className={cssLink} activeStyle={activeLinkStyle}>
          API
        </Link>
      </div>
      <div>footer</div>
    </div>
  );
}

Sidebar.propTypes = {
  siteTitle: PropTypes.string.isRequired,
};

export default Sidebar;
