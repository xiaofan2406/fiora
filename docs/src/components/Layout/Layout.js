import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';
import { theme, variables } from 'styles';
import 'styles/reset.css';
import 'styles/animation.css';

import Brand from './Brand';
import Navigation from './Navigation';

const layout = css`
  font-family: ${theme.fontFamily};
  font-size: ${theme.fontSize}px;
`;

const header = css`
  height: ${variables.Layout.headerHeight}px;
  background-color: ${theme.bgColor};
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

function Layout({ children }) {
  return (
    <div className={layout}>
      <div className={header}>
        <Brand />
        <Navigation height={100} />
      </div>
      {children}
    </div>
  );
}

Layout.propTypes = {
  children: PropTypes.node.isRequired
};

export { Layout as Component };

export default Layout;
