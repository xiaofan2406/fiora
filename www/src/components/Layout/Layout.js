import React from 'react';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';
import { css } from 'react-emotion';
import { MDXProvider } from '@mdx-js/tag';

import SiteMeta from './SiteMeta';
import Header from './Header';
import Sidebar from './Sidebar';
import PreComponent from './PreComponent';
import { cssPageWidth, headerHeight, sidebarWidth } from '../../styles';

function Layout({ children }) {
  return (
    <StaticQuery
      query={graphql`
        query SiteTitleQuery {
          site {
            siteMetadata {
              title
            }
          }
        }
      `}
    >
      {({ site: { siteMetadata } }) => (
        <>
          <SiteMeta {...siteMetadata} />
          <div
            className={css`
              min-height: calc(100vh - 18px);
              min-width: 480px;
              display: flex;
              flex-direction: column;
            `}
          >
            <Header title={siteMetadata.title} />
            <div
              className={css`
                ${cssPageWidth};
                margin: ${headerHeight}px 0 0 ${sidebarWidth}px;
                flex: 1;
                display: flex;
              `}
            >
              <Sidebar />
              <section
                className={css`
                  flex: 1;
                  padding: 24px;
                `}
              >
                <main
                  className={css`
                    width: 960px;
                    min-width: 960px;
                    margin: auto;
                  `}
                >
                  <MDXProvider components={{ pre: PreComponent }}>
                    {/* remove the extra div wrapping */}
                    <>{children.props.children}</>
                  </MDXProvider>
                </main>
              </section>
            </div>
          </div>
        </>
      )}
    </StaticQuery>
  );
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
