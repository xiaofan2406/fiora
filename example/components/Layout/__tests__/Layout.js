import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Component } from '../Layout';

it('should match snapshot', () => {
  const props = { classes: {}, children: '' };
  const component = shallow(<Component {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});

it('should render children', () => {
  const Child = () => <div>child</div>;
  const props = { classes: {}, children: <Child /> };
  const component = shallow(<Component {...props} />);

  expect(toJson(component)).toMatchSnapshot();
  expect(component.find(Child).length).toBe(1);
});
