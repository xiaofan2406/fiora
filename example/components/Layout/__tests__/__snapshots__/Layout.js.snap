// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="css-e2wgdg"
>
  <div
    className="css-1kqepsy"
  >
    <Brand />
    <Navigation
      height={100}
    />
  </div>
</div>
`;

exports[`should render children 1`] = `
<div
  className="css-e2wgdg"
>
  <div
    className="css-1kqepsy"
  >
    <Brand />
    <Navigation
      height={100}
    />
  </div>
  <Child />
</div>
`;
